src/shared/discover-image.c

```diff
@@ -282,6 +282,44 @@ static int extract_image_basename(
         return 0;
 }
 
+static int image_update_quota(Image *i, int fd) {
+        _cleanup_close_ int fd_close = -EBADF;
+        int r;
+
+        assert(i);
+
+        if (IMAGE_IS_VENDOR(i) || IMAGE_IS_HOST(i))
+                return -EROFS;
+
+        if (i->type != IMAGE_SUBVOLUME)
+                return -EOPNOTSUPP;
+
+        if (fd < 0) {
+                fd = open(i->path, O_CLOEXEC|O_NOCTTY|O_DIRECTORY);
```

Comment:
O_NOCTTY seems unnecessary when O_DIRECTORY is specified?

Reply:
O_NOCTTY is actually not necessary, but also used in `image_make()`. Let's keep the flag at least now for consistency.